ado/run_build.yml

```diff
@@ -0,0 +1,45 @@
+parameters:
+  targetPlatform: 'x64'
+
+jobs:
+- job: vs2019_hosted_${{ parameters.targetPlatform }}
+  pool:
+    name: Hosted Windows 2019 with VS2019
+    #name: Default
+
+  variables:
+    vcpkgLocation: '$(Build.SourcesDirectory)/vcpkg'
+    vcpkgResponseFile: $(Build.SourcesDirectory)/vcpkg_windows.txt
+  steps:
+    - checkout: self
+      submodules: recursive
+    - task: PowerShell@2
+      displayName: 'Install MSVC preview'
+      condition: or(contains(variables['Agent.Name'], 'Azure Pipelines'), contains(variables['Agent.Name'], 'Hosted Agent'))
+      inputs:
+        targetType: filePath
+        filePath: $(Build.SourcesDirectory)/ado/install_msvc_preview.ps1
+    - task: CacheBeta@0
+      displayName: Cache vcpkg
+      inputs:
+        key: $(vcpkgResponseFile) | $(Build.SourcesDirectory)/.git/modules/vcpkg/HEAD
+        path: '$(vcpkgLocation)'
+    - task: lucappa.cmake-ninja-vcpkg-tasks.d855c326-b1c0-4d6f-b1c7-440ade6835fb.run-vcpkg@0
```

Comment:
I don't understand what these tasks do, or why the need to have a uuid in their name?

Reply:
the run-vcpkg and run-cmake are two tasks contained in the [CppBuildTasks](https://marketplace.visualstudio.com/items?itemName=lucappa.cmake-ninja-vcpkg-tasks) ADO extension . Full instructions are there.

Yes, you could drop the Guids and just use instead

`- task run-vcpkg@0`

They are not stricly needed, you can directly write your own script that directly do a similar job.